autoscale/src/main/java/com/sequenceiq/periscope/monitor/ScalingHandlerUtil.java

```diff
@@ -0,0 +1,106 @@
+
+    @Inject
+    private Clock clock;
+
+    public void scaleIfNeed(Cluster cluster, BaseAlert alert) {
+        ScalingPolicy policy = alert.getScalingPolicy();
+        int totalNodes = Math.toIntExact(cloudbreakClient.autoscaleEndpoint().getHostMetadataCountForAutoscale(cluster.getStackId(), policy.getHostGroup()));
+        int desiredNodeCount = getDesiredNodeCount(cluster, policy, totalNodes);
+        if (totalNodes != desiredNodeCount) {
+            cluster.setLastScalingActivityCurrent();
+            clusterService.save(cluster);
+            scale(cluster, policy);
+        } else {
+            LOGGER.info("No scaling activity required for '{}' policy", policy.getName());
+        }
+    }
+
+    private void scale(Cluster cluster, ScalingPolicy policy) {
+        int totalNodes = Math.toIntExact(cloudbreakClient.autoscaleEndpoint().getHostMetadataCountForAutoscale(cluster.getStackId(), policy.getHostGroup()));
+        int desiredNodeCount = getDesiredNodeCount(cluster, policy, totalNodes);
+        Runnable scalingRequest = (Runnable) applicationContext.getBean("ScalingRequest", cluster, policy, totalNodes, desiredNodeCount);
+        loggedExecutorService.submit("ScalingHandler", scalingRequest);
+        rejectedThreadService.remove(cluster.getId());
+    }
+
+    public boolean isCooldownElapsed(Cluster cluster) {
+        long remainingTime = getRemainingCooldownTime(cluster);
+        if (remainingTime <= 0) {
+            return true;
+        }
+        LOGGER.info("Cluster cannot be scaled for {} min(s)",
+                ClusterUtils.TIME_FORMAT.format((double) remainingTime / TimeUtil.MIN_IN_MS));
+        return false;
+    }
+
+    private long getRemainingCooldownTime(Cluster cluster) {
+        long coolDown = cluster.getCoolDown();
+        long lastScalingActivity = cluster.getLastScalingActivity();
+        return lastScalingActivity == 0L ? 0L : (coolDown * TimeUtil.MIN_IN_MS) - (clock.getCurrentTime() - lastScalingActivity);
+    }
+
+    @VisibleForTesting
+    protected int getDesiredNodeCount(Cluster cluster, ScalingPolicy policy, int totalNodes) {
+        int scalingAdjustment = policy.getScalingAdjustment();
+        int desiredNodeCount;
+        switch (policy.getAdjustmentType()) {
+            case NODE_COUNT:
+                desiredNodeCount = totalNodes + scalingAdjustment;
+                break;
+            case PERCENTAGE:
+                desiredNodeCount = totalNodes
+                        + (int) (ceil(totalNodes * ((double) scalingAdjustment / ClusterUtils.MAX_CAPACITY)));
+                break;
+            case EXACT:
+                desiredNodeCount = policy.getScalingAdjustment();
+                break;
+            default:
+                desiredNodeCount = totalNodes;
+        }
+        int minSize = cluster.getMinSize();
+        int maxSize = cluster.getMaxSize();
+        return desiredNodeCount < minSize ? minSize : Math.min(desiredNodeCount, maxSize);
```

Comment:
pls add some extra logging to this including minSize, maxSize and calculated desiredNodeCount